src/OpenLoco/src/Vehicles/Vehicle.cpp

```diff
@@ -686,6 +686,88 @@ namespace OpenLoco::Vehicles
         return AirportObjectFlags::acceptsHeavyPlanes;
     }
 
+    // 0x004AFFF3
+    VehicleBogie* VehicleBogie::flipCar()
+    {
+        Vehicle train(head);
+        VehicleBase* precedingCarBody = train.veh2;
+        while (precedingCarBody->nextVehicleComponent()->asBase<VehicleBogie>() != this)
+        {
+            precedingCarBody = precedingCarBody->nextVehicleComponent();
+        }
+
+        VehicleBase* ptr = this; // lvalue required
+        CarComponent oldFirstComponent(ptr);
+        ptr = this; // CarComponent constructor changes value
+        Car car(ptr);
+
+        sfl::static_vector<CarComponent, VehicleObject::kMaxCarComponents> components;
+        for (CarComponent& component : car)
+        {
+            auto frontSprite = component.front->objectSpriteType;
+            component.front->objectSpriteType = component.back->objectSpriteType;
+            component.back->objectSpriteType = frontSprite;
+
+            // silence -Werror=null-dereference
+            VehicleBody* body = component.body;
+            body->var_38 ^= Flags38::isReversed;
+
+            components.push_back(component);
+        }
+        auto lastComponentIndex = components.size() - 1;
+        auto newFirstComponent = components.back();
+
+        // if the Car is only one CarComponent we don't have to swap any values
+        if (newFirstComponent.front == oldFirstComponent.front)
+        {
+            return this;
+        }
+
+        // silence -Werror=null-dereference
+        auto nextVehicleComponent = newFirstComponent.body->nextVehicleComponent();
```

Comment:
you don't have to dereference this you could just do `const auto nextVehicleComponentId = newFirstComponent.body->nextCarId;` Then you don't need the nullptr check after this.